res/controllers/Pioneer-DDJ-SR2-scripts.js

```diff
@@ -0,0 +1,1383 @@
+        wheelLedBaseChannel,
+        0x20+(channelOffset),
+        color
+    );
+};
+
+
+DDJSR2.pitchBendFromJog = function(group, movement) {
+    engine.setValue(group, "jog", movement / 5 * DDJSR2.jogwheelSensitivity);
+};
+
+DDJSR2.RingBufferView = class {
+    constructor(indexable, startIndex = 0) {
+        this.indexable = indexable;
+        this.index = startIndex;
+    }
+    advanceBy(n) {
+        this.index = script.posMod(this.index + n, this.indexable.length);
+        return this.current();
+    }
+    next() {
+        return this.advanceBy(1);
+    }
+    previous() {
+        return this.advanceBy(-1);
+    }
+    current() {
+        return this.indexable[this.index];
+    }
+};
+
+DDJSR2.BrowserContainer = function() {
+    const Browser = this;
+    this.browseKnob  = new components.ComponentContainer({
+        longPressTimer: 0,
+        longPressTimeout: 250,
+        trackColorCycleEnabled: false,
+        trackColorCycleHappened: false,
+        previewSeekEnabled: false,
+        previewSeekHappened: false,
+        turn: new components.Encoder({
+            group: "[Library]",
+            inKey: "MoveVertical",
+            inValueScale: DDJSR2.getRotaryDelta
+        }),
+        press: new components.Button({
+            group: "[Library]",
+            key: "GoToItem"
+        }),
+        shiftPress: new components.Button({
+            group: "[Library]",
+            input: function(channel, control, value, status, group) {
+                if (this.isPress(channel, control, value, status, group)) {
+                    Browser.browseKnob.trackColorCycleEnabled = true;
+                    return;
+                }
+                // On release, not press. Allows for track color cycling.
+                Browser.browseKnob.trackColorCycleEnabled = false;
+                if (Browser.browseKnob.trackColorCycleHappened) {
+                    Browser.browseKnob.trackColorCycleHappened = false;
+                    return;
+                }
+                if (!engine.getValue("[PreviewDeck1]", "play")) {
+                    engine.setValue("[PreviewDeck1]", "LoadSelectedTrackAndPlay", 1);
+                } else {
+                    script.triggerControl("[PreviewDeck1]", "stop");
+                }
+            }
+        }),
+        shiftTurn: new components.Encoder({
+            group: "[PreviewDeck1]",
+            input: function(channel, control, value, _status, _group) {
+                const rotateValue = DDJSR2.getRotaryDelta(value);
+                if (!Browser.browseKnob.trackColorCycleEnabled) {
+                    engine.setValue("[PreviewDeck1]", "beatjump", rotateValue*8);
+                } else {
+                    const key = (rotateValue > 0) ? "track_color_next" : "track_color_prev";
+                    engine.setValue("[Library]", key, 1.0);
+                    Browser.browseKnob.trackColorCycleHappened = true;
+                }
+            }
+        }),
```

Comment:
this should work just as well.
```suggestion
            input: function(channel, control, value, status, _group) {
                this.isPressed = this.isPress(channel, control, value, status);
                if (this.isPressed) {
                    return;
                }
                if (!engine.getValue("[PreviewDeck1]", "play")) {
                    engine.setValue("[PreviewDeck1]", "LoadSelectedTrackAndPlay", 1);
                } else {
                    script.triggerControl("[PreviewDeck1]", "stop");
                }
            }
        }),
        shiftTurn: new components.Encoder({
            beatJumpSize: 8,
            inValueScale: DDJSR2.getRotaryDelta,
            inSetValue: function(rotateValue) {
                if (Browser.browseKnob.isPressed) {
                    const direction = rotateValue > 0 ? "next" : "prev";
                    script.toggleControl("[Library]", `track_color_${direction}`);
                } else {
                    engine.setValue("[PreviewDeck1]", "beatjump", rotateValue*beatJumpSize;
                }
            }
        }),
```